samples/openapi3/client/petstore/dart-dio-next/petstore_client_lib_fake/doc/FakeApi.md

```diff
@@ -520,7 +520,7 @@ final int int32 = 56; // int | None
 final int int64 = 789; // int | None
 final double float = 3.4; // double | None
 final String string = string_example; // String | None
-final Uint8List binary = BINARY_DATA_HERE; // Uint8List | None
+final MultipartFile binary = BINARY_DATA_HERE; // MultipartFile | None
```

Comment:
This is probably not correct. That is why the special handling existed.

Reply:
I think dio should be able to handle all files as MultipartFile, no matter where

Reply:
This is not handled by `dio` this is handled by `built_value` serialization which doesn't have a serializer for `MultipartFile`. Not sure it works out of the box with `Uint8List` but one could add a custom Base64 serializer for it. This is not possible with `MultipartFile` as this is an abstraction around an async stream.

Reply:
I just noticed that there is a model "FormatTest" that has a `binary` property.
and looking at `csharp-netcore`, they have these type mappings
```java
typeMapping.put("file", "System.IO.Stream");
typeMapping.put("binary", "byte[]");
```
which for our case should be
```java
typeMapping.put("file", "MultipartFile");
typeMapping.put("binary", "Uint8List");
```

Reply:
Yea but I think a we also have a form param in the sample which is `binary`, not sure I remember that correctly.

Reply:
actually looking at it more:
`csharp-netcore` also generates `System.IO.Stream` for `binary` formats (seems that `binary` aliases into `file` for backward compatibility I think?)
https://github.com/OpenAPITools/openapi-generator/blob/master/samples/client/petstore/csharp-netcore/OpenAPIClient-net5.0/src/Org.OpenAPITools/Model/FormatTest.cs#L143

which is also not correct, and looking at OAS3 specification: https://swagger.io/docs/specification/describing-responses/#response-that-returns-a-file

it appears that the `FormatTest` model itself is wrong here, as having an embedded property that returns `format: binary` is illegal.

So that leaves us with responses with ONLY a file as a response, and in that case the response shouldn't get de-serialized by anything at all, and instead let dio return the file stream.

similar to the download method here:
https://github.com/flutterchina/dio/blob/20506b496d41d7c2cc71d323e22634e99dc3ef21/dio/lib/src/entry/dio_for_native.dart#L84

Reply:
@wing328 do you think this schema is in line with the specification?  https://github.com/OpenAPITools/openapi-generator/blob/158649aff9a87b13b448880d7ca6a4ac35fadcc2/modules/openapi-generator/src/test/resources/3_0/petstore-with-fake-endpoints-models-for-testing.yaml#L1510

Reply:
`file` is the legacy/OAS2 type and gets mapped to `binary` now or the other way around. 

> it appears that the FormatTest model itself is wrong here, as having an embedded property that returns format: binary is illegal.

This is probably right, I am all for removing this from the sample and replace it (if it is not already there) with `format: byte`. This should be done in a separate PR.

> So that leaves us with responses with ONLY a file as a response, and in that case the response shouldn't get de-serialized by anything at all, and instead let dio return the file stream.

Yes, we could do that and return a stream, ideally we give the user both options. One with `ResponseType.byte` and one `$operationDownload` with `ResponseType.stream`.  But this should also be a separate PR.